### PR TITLE
Cleanup sys module to match house style

### PR DESCRIPTION
**Pull request by ChrisDenton**
This moves a test file out of sys as it's just testing std types. Also cleans up some assorted bits including making the `use` statements match the house style.

### CODE REVIEW

File: library/std/src/sys/anonymous_pipe/windows.rs [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+    AsHandle, AsRawHandle, BorrowedHandle, FromRawHandle, IntoRawHandle, OwnedHandle, RawHandle,
 };
+use crate::pipe::{PipeReader, PipeWriter};
+use crate::process::Stdio;
+use crate::sys::c;
+use crate::sys::handle::Handle;
+use crate::sys_common::{FromInner, IntoInner};
 
-pub(crate) type AnonPipe = Handle;
+use core::ptr;
```
Comment (joboet) [inline library/std/src/sys/anonymous_pipe/windows.rs:11]:
```suggestion
use crate::ptr;
```

**Review by joboet — APPROVED** [review]
Thanks!
r=me with the nit (and the corresponding reformatting) applied

**Comment by ChrisDenton** [comment]
I had to rebase due to changes in how imports are formatted. However the code is otherwise the same as before the rebase.

@bors r=joboet

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang/rust/pull/128428#issuecomment-2260159956

**Comment by ChrisDenton** [comment]
I left in a `pub(crate)` in the unsupported implementation, oops.

@bors r=joboet

**Pull request merged**